### DISCUSSION TITLE
AWS Secrets Manager support

### DISCUSSION THREAD

**Discussion by beastlyDartfordian** [OrchardCMS/OrchardCore#15491] [Ideas]
We use AWS for our hosting here at our company, and with that use Secrets Manager to store, well, secrets. 

Wondering what the thoughts / potential was for adding secrets manager support similar to what Orchard Core has for Azure Key Vault?

(If its already possible to use AWS Secrets Manager but I don't know about it please let me know!)

Cheers!

**Comment by Piedone** [comment]
Just as we [support S3 for Media storage](https://docs.orchardcore.net/en/latest/docs/reference/modules/Media.AmazonS3/) and [Azure Key Vault](https://docs.orchardcore.net/en/latest/docs/reference/core/KeyVault.Azure/), we could support AWS Secrets Manager, but we don't, currently.

I'm not too familiar with AWS Secrets Manager though, and perhaps such an explicit support is not needed. How exactly would you like to manage secrets?

Because if it's only about storing configuration values of the app like connection strings, then (similar to Azure Key Vault) it might be able to inject those as environment variables, making them readily accessible from the app via the default ASP.NET Core configuration provider.
